src/Management/src/Diagnostics/EventPipe/IpcClient.cs

```diff
@@ -0,0 +1,115 @@
+﻿// <auto-generated/>
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.IO.Pipes;
+using System.Linq;
+using System.Net.Sockets;
+using System.Runtime.InteropServices;
+using System.Security.Principal;
+using System.Text;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.Diagnostics.Tools.RuntimeClient.DiagnosticsIpc
+{
+    public class IpcClient
+    {
+        private static string DiagnosticsPortPattern { get; } = RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? @"^dotnet-diagnostic-(\d+)$" : @"^dotnet-diagnostic-(\d+)-(\d+)-socket$";
+
+        private static string IpcRootPath { get; } = RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? @"\\.\pipe\" : Path.GetTempPath();
+
+        private static double ConnectTimeoutMilliseconds { get; } = TimeSpan.FromSeconds(3).TotalMilliseconds;
+
+        /// <summary>
+        /// Get the OS Transport to be used for communicating with a dotnet process.
+        /// </summary>
+        /// <param name="processId">The PID of the dotnet process to get the transport for</param>
+        /// <returns>A System.IO.Stream wrapper around the transport</returns>
+        private static Stream GetTransport(int processId)
+        {
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                string pipeName = $"dotnet-diagnostic-{processId}";
+                var namedPipe = new NamedPipeClientStream(
+                    ".", pipeName, PipeDirection.InOut, PipeOptions.None, TokenImpersonationLevel.Impersonation);
+                namedPipe.Connect((int)ConnectTimeoutMilliseconds);
+                return namedPipe;
+            }
+            else
+            {
+                string ipcPort = Directory.GetFiles(IpcRootPath) // Try best match.
+                    .Select(namedPipe => (new FileInfo(namedPipe)).Name)
+                    .SingleOrDefault(input => Regex.IsMatch(input, $"^dotnet-diagnostic-{processId}-(\\d+)-socket$"));
+                if (ipcPort == null)
+                {
+                    throw new PlatformNotSupportedException($"Process {processId} not running compatible .NET Core runtime");
+                }
+                string path = Path.Combine(Path.GetTempPath(), ipcPort);
+                // TODO: Enable this for appropriate build
+                // var remoteEP = new UnixDomainSocketEndPoint(path);
```

Comment:
With the version bumped up, can we try and see if this compile?
This line is critical to access the pipe.

Reply:
Yes, please go ahead and change anything you think needs changing here